packages/core/src/components/utils/theme-switcher.ts

```diff
@@ -0,0 +1,84 @@
+import { TypedEvent } from './typed-event';
+
+export class ThemeSwitcher {
```

Comment:
I think it is not necessary to handle the theme switcher as a class only my suggestions would be:

```
export class ThemeSwitcher { ... }

const themeSwitcher = new ThemeSwitcher();
export default themeSwitcher;
```